ui/src/app/admin/_components/personTable.tsx

```diff
@@ -0,0 +1,284 @@
+'use client';
+
+import {
+    useReactTable,
+    flexRender,
+    getCoreRowModel,
+    getPaginationRowModel,
+    getFilteredRowModel,
+    getSortedRowModel,
+    SortingState,
+    RowSelectionState
+} from '@tanstack/react-table';
+import { Table, TableBody, TableCell, TableHead, TableHeader, TableRow } from '@/components/ui/table';
+import PaginationBar from '@/components/table/table-element/pagination-bar';
+import GlobalFilter from '@/components/table/table-element/global-filter';
+import SortArrow from '@/components/table/table-element/sort-arrow';
+import { useState } from 'react';
+import { ArrowUpRightFromSquare } from 'lucide-react';
+import { CrownIcon } from 'lucide-react';
+import { Button } from '@/components/ui/button';
+import SearchInput from '@/app/admin/_components/searchInput';
+import personTableColumns from '@/components/table/table-element/person-table-columns';
+import PersonTableTooltip from '@/app/admin/_components/personTableTooltip';
+import Link from 'next/link';
+import { useRouter } from 'next/navigation';
+import { groupingOwners, removeFromGroups } from '@/lib/actions';
+import OwnersModal from '@/components/modal/owners-modal';
+import { Alert, AlertDescription } from '@/components/ui/alert';
+
+const pageSize = parseInt(process.env.NEXT_PUBLIC_PAGE_SIZE as string);
+
+const PersonTable = (data) => {
+    const [globalFilter, setGlobalFilter] = useState('');
+    const [sorting, setSorting] = useState<SortingState>([]);
+    const [rowSelection, setRowSelection] = useState<RowSelectionState>({});
+    const [modalOpen, setModalOpen] = useState(false);
+    const [modalData, setModalData] = useState([]);
+    const [dummyBool, setDummyBool] = useState(false);
+    const router = useRouter();
+    const searchUid = data.searchUid;
+    const validUid = data.groupingsInfo.resultCode;
+    const groupingsInfo = data.groupingsInfo.results;
+    const userInfo = data.userInfo;
+    const hydrateModal = async (path) => {
+        setModalData((await groupingOwners(path)).members);
+        setModalOpen(true);
+    };
+
+    const close = () => {
+        setModalOpen(false);
+    };
+
+    const handleRemove = async () => {
+        const numSelected = table.getSelectedRowModel().rows.length;
+        const arr = [];
+        let i;
+        for (i = 0; i < numSelected; i++) {
+            const original = table.getSelectedRowModel().rows[i].original;
+            if (original.inOwner) arr.push(original.path + ':owners');
+            if (original.inInclude) arr.push(original.path + ':include');
+            if (original.inExclude) arr.push(original.path + ':exclude');
+        }
```

Comment:
There's a very nice way to rewrite this code using the .flatMap function: (Generated by ChatGPT!)
```
const groups = table.getSelectedRowModel().rows.flatMap(({ original }) => {
    const { path, inOwner, inInclude, inExclude } = original;
    return [
        inOwner && `${path}:owners`,
        inInclude && `${path}:include`,
        inExclude && `${path}:exclude`,
    ].filter(groupPath => groupPath);
});
```